[BUG] Multi-gpu training notebook is giving error if we generate schema from core

### Bug description
<!-- A clear and concise description of what the bug is. -->

I am getting the following error when I run multi-gpu training [notebook](https://github.com/NVIDIA-Merlin/Transformers4Rec/blob/main/examples/end-to-end-session-based/03-Session-based-Yoochoose-multigpu-training-PyT.ipynb)

```
/usr/local/lib/python3.8/dist-packages/torch/distributed/launch.py:180: FutureWarning: The module torch.distributed.launch is deprecated
and will be removed in future. Use torchrun.
Note that --use_env is set by default in torchrun.
If your script expects `--local_rank` argument to be set, please
change it to read from `os.environ['LOCAL_RANK']` instead. See 
https://pytorch.org/docs/stable/distributed.html#launch-utility for 
further instructions

  warnings.warn(
WARNING:torch.distributed.run:
*****************************************
Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed. 
*****************************************
/usr/local/lib/python3.8/dist-packages/merlin/dtypes/mappings/tf.py:52: UserWarning: Tensorflow dtype mappings did not load successfully due to an error: No module named 'tensorflow'
  warn(f"Tensorflow dtype mappings did not load successfully due to an error: {exc.msg}")
/usr/local/lib/python3.8/dist-packages/merlin/dtypes/mappings/tf.py:52: UserWarning: Tensorflow dtype mappings did not load successfully due to an error: No module named 'tensorflow'
  warn(f"Tensorflow dtype mappings did not load successfully due to an error: {exc.msg}")
Traceback (most recent call last):
  File "pyt_trainer.py", line 41, in <module>
    input_module = tr.TabularSequenceFeatures.from_schema(
  File "/usr/local/lib/python3.8/dist-packages/transformers4rec/torch/features/sequence.py", line 193, in from_schema
    output: TabularSequenceFeatures = super().from_schema(  # type: ignore
  File "/usr/local/lib/python3.8/dist-packages/transformers4rec/torch/features/tabular.py", line 176, in from_schema
    output = cls(
  File "/usr/local/lib/python3.8/dist-packages/transformers4rec/torch/features/sequence.py", line 127, in __init__
    super().__init__(
  File "/usr/local/lib/python3.8/dist-packages/transformers4rec/torch/features/tabular.py", line 84, in __init__
    assert to_merge != {}, "Please provide at least one input layer"
AssertionError: Please provide at least one input layer
Traceback (most recent call last):
  File "pyt_trainer.py", line 41, in <module>
    input_module = tr.TabularSequenceFeatures.from_schema(
  File "/usr/local/lib/python3.8/dist-packages/transformers4rec/torch/features/sequence.py", line 193, in from_schema
    output: TabularSequenceFeatures = super().from_schema(  # type: ignore
  File "/usr/local/lib/python3.8/dist-packages/transformers4rec/torch/features/tabular.py", line 176, in from_schema
    output = cls(
  File "/usr/local/lib/python3.8/dist-packages/transformers4rec/torch/features/sequence.py", line 127, in __init__
    super().__init__(
  File "/usr/local/lib/python3.8/dist-packages/transformers4rec/torch/features/tabular.py", line 84, in __init__
    assert to_merge != {}, "Please provide at least one input layer"
AssertionError: Please provide at least one input layer
ERROR:torch.distributed.elastic.multiprocessing.api:failed (exitcode: 1) local_rank: 0 (pid: 23905) of binary: /usr/bin/python
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.8/dist-packages/torch/distributed/launch.py", line 195, in <module>
    main()
  File "/usr/local/lib/python3.8/dist-packages/torch/distributed/launch.py", line 191, in main
    launch(args)
  File "/usr/local/lib/python3.8/dist-packages/torch/distributed/launch.py", line 176, in launch
    run(args)
  File "/usr/local/lib/python3.8/dist-packages/torch/distributed/run.py", line 753, in run
    elastic_launch(
  File "/usr/local/lib/python3.8/dist-packages/torch/distributed/launcher/api.py", line 132, in __call__
    return launch_agent(self._config, self._entrypoint, list(args))
  File "/usr/local/lib/python3.8/dist-packages/torch/distributed/launcher/api.py", line 246, in launch_agent
    raise ChildFailedError(
torch.distributed.elastic.multiprocessing.errors.ChildFailedError: 
============================================================
pyt_trainer.py FAILED
------------------------------------------------------------
Failures:
[1]:
  time      : 2023-03-16_19:30:24
  host      : 1902e905751e
  rank      : 1 (local_rank: 1)
  exitcode  : 1 (pid: 23906)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
------------------------------------------------------------
Root Cause (first observed failure):
[0]:
  time      : 2023-03-16_19:30:24
  host      : 1902e905751e
  rank      : 0 (local_rank: 0)
  exitcode  : 1 (pid: 23905)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
```

### Steps/Code to reproduce bug

You need to run `01` and `03` notebooks in this [folder](https://github.com/NVIDIA-Merlin/Transformers4Rec/tree/main/examples/end-to-end-session-based) in order. For dataset generation you can use 

```
import calendar
import datetime

import numpy as np
import pandas as pd


def generate_synthetic_data(
    start_date: datetime.date, end_date: datetime.date, rows_per_day: int = 1000
) -> pd.DataFrame:
    assert end_date > start_date, "end_date must be later than start_date"

    number_of_days = (end_date - start_date).days
    total_number_of_rows = number_of_days * rows_per_day

    # Generate a long-tail distribution of item interactions. This simulates that some items are
    # more popular than others.
    long_tailed_item_distribution = np.clip(
        np.random.lognormal(3.0, 1.0, total_number_of_rows).astype(np.int64), 1, 50000
    )

    # generate random item interaction features
    df = pd.DataFrame(
        {
            "session_id": np.random.randint(70000, 80000, total_number_of_rows),
            "item_id": long_tailed_item_distribution,
        },
    )

    # generate category mapping for each item-id
    df["category"] = pd.cut(df["item_id"], bins=334, labels=np.arange(1, 335)).astype(
        np.int64
    )

    max_session_length = 60 * 60  # 1 hour

    def add_timestamp_to_session(session: pd.DataFrame):
        random_start_date_and_time = calendar.timegm(
            (
                start_date
                # Add day offset from start_date
                + datetime.timedelta(days=np.random.randint(0, number_of_days))
                # Add time offset within the random day
                + datetime.timedelta(seconds=np.random.randint(0, 86_400))
            ).timetuple()
        )
        session["timestamp"] = random_start_date_and_time + np.clip(
            np.random.lognormal(3.0, 1.0, len(session)).astype(np.int64),
            0,
            max_session_length,
        )
        return session

    df = df.groupby("session_id").apply(add_timestamp_to_session).reset_index()

    return df

interactions_df = generate_synthetic_data(datetime.date(2014, 4, 1), datetime.date(2014, 6, 30))
interactions_df = cudf.from_pandas(interactions_df)
```

### Environment details
- Transformers4Rec version: 
- Platform:
- Python version:
- Huggingface Transformers version:
- PyTorch version (GPU?):
- Tensorflow version (GPU?):

### Additional context
I am using `merlin-pytorch:23.02` image with the latest main branches pulled from libs.

--- a/examples/end-to-end-session-based/03-Session-based-Yoochoose-multigpu-training-PyT.ipynb
+++ b/examples/end-to-end-session-based/03-Session-based-Yoochoose-multigpu-training-PyT.ipynb
@@ -172,12 +172,12 @@
     "            max_sequence_length=20,\n",
     "            data_loader_engine='merlin',\n",
     "            num_train_epochs=10, \n",
-    "            dataloader_drop_last=False,\n",
+    "            dataloader_drop_last=True,\n",
     "            per_device_train_batch_size = sh_args.per_device_train_batch_size,\n",
     "            per_device_eval_batch_size = sh_args.per_device_eval_batch_size,\n",
     "            learning_rate=sh_args.learning_rate,\n",
     "            report_to = [],\n",
-    "            logging_steps=200\n",
+    "            logging_steps=200,\n",
     "        )\n",
     "\n",
     "# Instantiate the trainer\n",
@@ -204,7 +204,6 @@
     "    time_index_eval = time_index + 1\n",
     "    train_paths = glob.glob(os.path.join(OUTPUT_DIR, f\"{time_index_train}/train.parquet\"))\n",
     "    eval_paths = glob.glob(os.path.join(OUTPUT_DIR, f\"{time_index_eval}/valid.parquet\"))\n",
-    "    print(train_paths)\n",
     "    \n",
     "    # Train on day related to time_index \n",
     "    print('*'*20)\n",
@@ -215,7 +214,7 @@
     "    recsys_trainer.train()\n",
     "    recsys_trainer.state.global_step +=1\n",
     "    print('finished')\n",
-    "    \n",
+    "\n",
     "    # Evaluate on the following day\n",
     "    recsys_trainer.eval_dataset_or_path = eval_paths\n",
     "    train_metrics = recsys_trainer.evaluate(metric_key_prefix='eval')\n",
@@ -227,7 +226,7 @@
     "    wipe_memory()\n",
     "\n",
     "end = time.time()\n",
-    "print('Total training time:',end-start)\n"
+    "print('Total training time:',end-start)"
    ]
   },
   {
@@ -259,7 +258,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "!python -m torch.distributed.launch --nproc_per_node 2 pyt_trainer.py --path \"/workspace/data/preproc_sessions_by_day\" --learning-rate 0.0005"
+    "! torchrun --nproc_per_node 2 pyt_trainer.py --path \"/workspace/data/preproc_sessions_by_day\" --learning-rate 0.0005"
    ]
   },
   {

--- a/transformers4rec/torch/trainer.py
+++ b/transformers4rec/torch/trainer.py
@@ -128,6 +128,14 @@ class Trainer(BaseTrainer):
         self.schema = schema
         self.incremental_logging = incremental_logging
 
+        # Set global_rank and global_size if DDP is used
+        if self.args.local_rank != -1:
+            self.device = self.local_rank = self.args.local_rank
+            self.global_size = self.args.world_size
+        else:
+            self.device = self.local_rank = None
+            self.global_size = None
+
     def get_train_dataloader(self):
         """
         Set the train dataloader to use by Trainer.
@@ -140,14 +148,6 @@ class Trainer(BaseTrainer):
 
         assert self.schema is not None, "schema is required to generate Train Dataloader"
 
-        # Set global_rank and global_size if DDP is used
-        if self.args.local_rank != -1:
-            local_rank = self.args.local_rank
-            global_size = self.args.world_size
-        else:
-            local_rank = None
-            global_size = None
-
         return T4RecDataLoader.parse(self.args.data_loader_engine).from_schema(
             self.schema,
             self.train_dataset_or_path,
@@ -156,8 +156,9 @@ class Trainer(BaseTrainer):
             drop_last=self.args.dataloader_drop_last,
             shuffle=True,
             shuffle_buffer_size=self.args.shuffle_buffer_size,
-            global_rank=local_rank,
-            global_size=global_size,
+            global_rank=self.local_rank,
+            global_size=self.global_size,
+            device=self.device,
         )
 
     def get_eval_dataloader(self, eval_dataset=None):
@@ -174,6 +175,7 @@ class Trainer(BaseTrainer):
             raise ValueError("Trainer: evaluation requires an eval_dataset.")
         eval_dataset = eval_dataset if eval_dataset is not None else self.eval_dataset
         assert self.schema is not None, "schema is required to generate Eval Dataloader"
+
         return T4RecDataLoader.parse(self.args.data_loader_engine).from_schema(
             self.schema,
             self.eval_dataset_or_path,
@@ -182,6 +184,9 @@ class Trainer(BaseTrainer):
             drop_last=self.args.dataloader_drop_last,
             shuffle=False,
             shuffle_buffer_size=self.args.shuffle_buffer_size,
+            global_rank=self.local_rank,
+            global_size=self.global_size,
+            device=self.device,
         )
 
     def get_test_dataloader(self, test_dataset=None):
@@ -206,6 +211,9 @@ class Trainer(BaseTrainer):
             drop_last=self.args.dataloader_drop_last,
             shuffle=False,
             shuffle_buffer_size=self.args.shuffle_buffer_size,
+            global_rank=self.local_rank,
+            global_size=self.global_size,
+            device=self.device,
         )
 
     def num_examples(self, dataloader: DataLoader):

